requests/utils.py

```diff
@@ -665,6 +665,21 @@ def parse_header_links(value):
 
     return links
 
+def is_valid_location(response):
+    """Verify that multiple Location headers weren't
+    returned from the last response.
+    """
+    if hasattr(response.raw, 'headers'):
+        if hasattr(response.raw.headers, 'getlist'):
+            return len(response.raw.headers.getlist('location')) <= 1
+        else:
+            pass # Do something with header value
+    elif hasattr(response.raw, 'read'):
+        response.raw.seek(0) # Move to beginning of file
+        data = response.raw.read()
+        # parse_headers(data) # try to parse headers out of raw response
```

Comment:
Going through a few libraries that implement different adapters, the `raw` field seems considerably varied in it's use. 

In the event we end up with something file-like, such as `StringIO`, we can parse the headers out, but only if it's the complete response. In the limited testing I did though, this was typically just the response body. I'm not sure if we can do anything useful in that case. Also if it is the response body and happens to contain contents mimicking an HTTP response, then we could in theory flag this incorrectly.

I'm just seeing a lot of unknowns here that I'm not sure have a clean fix. Perhaps just returning True when we can't make a reasonable discernment would be best.